javascript_nodejs/4.simple-prompt-bot/index.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const restify = require('restify');
+const path = require('path');
+const ERROR = 1;
+
+// Import reuqired bot services. See https://ama.ms/bot-services to learn more about the different part of a bot
+const { BotFrameworkAdapter, MemoryStorage, ConversationState } = require('botbuilder');
+const { BotConfiguration } = require('botframework-config');
+
+const MainDialog = require('./dialogs/mainDialog');
+
+// Read botFilePath and botFileSecret from .env file
+// Note: Ensure you have a .env file and include botFilePath and botFileSecret.
+const ENV_FILE = path.join(__dirname, '.env');
+const env = require('dotenv').config({path: ENV_FILE});
+
+// Create HTTP server
+let server = restify.createServer();
+server.listen(process.env.port || process.env.PORT || 3978, function () {
+    console.log(`\n${server.name} listening to ${server.url}`);
+    console.log(`\nGet Bot Framework Emulator: https://aka.ms/botframework-emulator`);
+    console.log(`\nTo talk to your bot, open simple-prompt-bot.bot file in the Emulator`);
+});
+
+// .bot file path
+const BOT_FILE = path.join(__dirname, (process.env.botFilePath || ''));
+
+console.log('reading config from ', BOT_FILE);
+// read bot configuration from .bot file. 
+let botConfig;
+try {
+    botConfig = BotConfiguration.loadSync(BOT_FILE, process.env.botFileSecret);
+} catch (err) {
+    console.log(`Error reading bot file. Please ensure you have valid botFilePath and botFileSecret set for your environment`);
+    console.log(err);
+    process.exit(ERROR);
+}
+
+// bot name as defined in .bot file 
+// See https://aka.ms/about-bot-file to learn more about .bot file its use and bot configuration .
+const BOT_CONFIGURATION = 'simple-prompt-bot';
+
+// Get bot endpoint configuration by service name
+const endpointConfig = botConfig.findServiceByNameOrId(BOT_CONFIGURATION);
+
+// Create adapter. See https://aka.ms/about-bot-adapter to learn more about .bot file its use and bot configuration .
+const adapter = new BotFrameworkAdapter({
+    appId: endpointConfig.appId || process.env.microsoftAppID,
+    appPassword: endpointConfig.appPassword || process.env.microsoftAppPassword
+});
+
+// Define state store for your bot. See https://aka.ms/about-bot-state to learn more about bots memory service
+// A bot requires a sate store to priciest it dialog and user state between messages
```

Comment:
spelling. "sate" != "state". 
Grammer. 
"priciest"? Not sure what's intended with that one.

Reply:
priciest != persist.